src/services/storage-options/restApiService.js

```diff
@@ -99,15 +99,32 @@ export default class RestApiService extends StorageInterface {
   async updateConfig(config) {
     const response = await axios.put(
       `http://localhost:5000/v1/app/org/f352ae63-11fc-4dbe-bab1-72561aa25fca/config/5e0f71ff-987d-4240-85eb-df6adf568c31`,
-      { config }
+      { config, withCredentials: true }
     );
     return response.data.config;
   }
 
   async getCredentials() {
     const { data } = await axios.get(
-      "http://localhost:5000/v1/app/profile/token"
+      "http://localhost:5000/v1/app/profile/token",
+      {
+        withCredentials: true,
+      }
     );
+    const allCookies = document.cookie;
+    const cookieArray = allCookies.split("; ");
+    let accessToken = null;
+    for (const cookie of cookieArray) {
+      const [name, value] = cookie.split("=");
+      if (name.trim() === "access_token") {
+        accessToken = value;
+        break;
+      }
+    }
+
+    if (accessToken) data.type = "cookie";
+    else data.type = "bearer";
```

Comment:
This is wrong.

You should only call the `app/profile/token` endpoint if a cookie doesn't exist.  That endpoint generates a new bearer token.

So you should check for cookies first and build `data` from the values in the cookie/JWT when it exists.  Then if it doesn't exist, use the token endpoint to create a bearer token.

Reply:
@dacoaster. 
Ok, I got it.
I will update the code and let you know when it's done.

Reply:
@dacoaster , if I use the accessToken of cookie, how can I get oauthTokenIds, loggedInAt and user data?

Reply:
You can skip all of these fields for now.

Reply:
Ok, I got it.